it_zwierzak/user/views.py

```diff
@@ -1,45 +1,121 @@
 from django.shortcuts import render, redirect
-from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth import authenticate, login as auth_login, logout as auth_logout, update_session_auth_hash
 from django.contrib import messages
-
-from .forms import CreateUserForm
+from django.contrib.auth.decorators import login_required
+from user.models import Profile
+from .forms import CreateUserForm, ProfileForm, ChangePasswordForm
 
 
 def register(request):
-    form = CreateUserForm()
+    if request.user.is_authenticated:
+        return redirect('homepage:home')
+    else:
+        form = CreateUserForm()
 
-    if request.method == 'POST':
-        form = CreateUserForm(request.POST)
-        if form.is_valid():
-            form.save()
-            username = form.cleaned_data.get('username')
-            messages.success(request, f'Utworzono profil dla: {username}')
+        if request.method == 'POST':
+            form = CreateUserForm(request.POST)
+            if form.is_valid():
+                form.save()
+                username = form.cleaned_data.get('username')
+                messages.success(request, f'Utworzono profil dla: {username}')
 
-            return redirect('user:login')
+                return redirect('user:login')
 
-    context = {'form': form}
-    return render(
-        request,
-        'user/register.html',
-        context
-    )
+        context = {'form': form}
+        return render(
+            request,
+            'user/register.html',
+            context
+            )
 
 
 def login(request):
+    if request.user.is_authenticated:
+        return redirect('homepage:home')
+    else:
+        if request.method == 'POST':
+            username = request.POST.get('username')
+            password = request.POST.get('password')
+
+            user = authenticate(request, username=username, password=password)
+
+            if user is not None:
+                auth_login(request, user)
+                return redirect('homepage:home')
+            else:
+                messages.info(request, 'Nazwa użytkownika lub hasło jest niepoprawne')
+
+        return render(
+            request,
+            'user/login.html',
+        )
+
+
+def logout(request):
+    auth_logout(request)
+    return redirect('user:login')
 
-    if request.method == 'POST':
-        username = request.POST.get('username')
-        password = request.POST.get('password')
 
-        user = authenticate(request, username=username, password=password)
+@login_required(login_url='login')
+def profile(request):
 
-        if user is not None:
-            login(request, user)
-            return redirect('homepage:home')
+    if request.method == "GET":
+        form = ProfileForm()
+        return render(request, 'user/user_profile.html', {'form': form})
```

Comment:
To takie jeszcze jedno usprawnienie, jestem już użytkownikiem co ma profil, chcę jedną rzecz zmienić tylko, teraz będę musiał podawać wszystko od nowa w formularzu jeśli dobrze tu widzę, fajnie byłoby załadować już dane które mamy dostępne jeśli takie mamy